File: User-Frontend-Service/src/services/TrainingPlanService.js
import axiosInstance from '../services/AxiosInstance';
import { store } from '../store/store';
import React, { useState } from 'react'

const state = store.getState();


export function addNewTrainingPlan(trainingPlan, userId){
    return axiosInstance.post(`/training-plans/${userId}`, trainingPlan);
}

export function getAllUserTrainingPlans(userId){
    const url = `/training-plans/${userId}/all`
    return axiosInstance.get(url)
}

export function getTrainingPlan(trainingPlanId){
    const url = `/training-plans/${trainingPlanId}`
    return axiosInstance.get(url)
}

export function updateTrainingPlanDetails(trainingPlanId, trainingPlanDetails){
    const url = `/training-plans/${trainingPlanId}`
    return axiosInstance.put(url, trainingPlanDetails)
}

export function deleteTrainingUnitPart(trainingPlanId, trainingUnitId, trainingUnitPartId){
    const url = `/training-plans/${trainingPlanId}/training-units/${trainingUnitId}/unit-parts/${trainingUnitPartId}`
    return axiosInstance.delete(url)
}

export function addTrainingUnitPart(trainingPlanId, trainingUnitId, createTrainingUnitPartRequest){
    const url = `/training-plans/${trainingPlanId}/training-units/${trainingUnitId}`
    return axiosInstance.post(url, createTrainingUnitPartRequest)
}

export function updateTrainingUnitPart(trainingPlanId,  trainingUnitId, trainingUnitPartId, updateTrainingUnitPartRequest){
    const url = `/training-plans/${trainingPlanId}/training-units/${trainingUnitId}/unit-parts/${trainingUnitPartId}`
    return axiosInstance.put(url, updateTrainingUnitPartRequest)
}

export function getCurrentTrainingPlan(userId){
    const url = `/users/${userId}/training-plan`
    return axiosInstance.get(url)
}

export function setCurrentUserTrianingPlan(userId, trainingPlanId){
    const url = `/users/${userId}/training-plan/${trainingPlanId}`
    return axiosInstance.put(url)
}

export function getAllPublicTrainingPlans(userId){
    const url = `/training-plans/public?requestedByUserId=${userId}`
    return axiosInstance.get(url)
}

export function copyTrainingPlan(trainingPlanId, forUserId){
    const url = `/training-plans/${trainingPlanId}/copy/${forUserId}`
    return axiosInstance.post(url)
}

export function addTrainingPlanRate(trainingPlanId, request){
    const url = `/training-plans/${trainingPlanId}/rates`
    return axiosInstance.post(url, request)
}

File: Training-Plan-Configuration-Service/src/main/java/pm/workout/helper/domain/training/plan/TrainingPlanService.java
package pm.workout.helper.domain.training.plan;

import pm.workout.helper.api.training.plan.request.CreateTrainingPlanRequest;
import pm.workout.helper.api.training.plan.request.CreateTrainingUnitPartRequest;
import pm.workout.helper.api.training.plan.request.UpdateTrainingPlanRequest;
import pm.workout.helper.api.training.plan.request.UpdateTrainingUnitPartRequest;
import pm.workout.helper.api.training.plan.request.plan.AddTrainingDayRequest;
import pm.workout.helper.api.training.plan.request.plan.AddTrainingPlanRateRequest;
import pm.workout.helper.domain.training.plan.dto.PublicTrainingPlanDto;
import pm.workout.helper.domain.training.plan.dto.TrainingPlanDetailsDto;
import pm.workout.helper.domain.training.plan.dto.TrainingPlanDto;
import pm.workout.helper.domain.training.plan.dto.template.TrainingUnitTemplateDto;

import java.util.List;

public interface TrainingPlanService {
    List<TrainingPlanDto> getAllUserTrainingPlans(Long userId);
    TrainingPlanDetailsDto getTrainingPlanDetails(Long trainingPlanId);
    Long createNewTrainingPlan(CreateTrainingPlanRequest request, Long userId);
    void deleteTrainingPlan(Long trainingPlanId);
    TrainingUnitTemplateDto getTrainingUnitPartTemplate(Long trainingPlanId, TrainingDay trainingDay);
    void updateTrainingPlan(Long trainingPlanId, UpdateTrainingPlanRequest updateTrainingPlanRequest);
    void deleteTrainingUnitPart(Long trainingPlanId, Long trainingUnitId, Long trainingUnitPartId);
    void addTrainingUnitPart(Long trainingPlanId, Long trainingUnitId, CreateTrainingUnitPartRequest request);
    void updateTrainingUnitPart(Long trainingPlanId, Long trainingUnitId, Long trainingUnitPartId, UpdateTrainingUnitPartRequest request);
    List<PublicTrainingPlanDto> getAllPublicTrainingPlans(Long requestedByUserId);
    void copyTrainingPlan(Long trainingPlanId, Long userId);
    void addTrainingPlanRate(Long trainingPlanId, AddTrainingPlanRateRequest request);
}
